modal/cli/import_refs.py

```diff
@@ -31,51 +32,66 @@
 @dataclasses.dataclass
 class ImportRef:
     file_or_module: str
+    is_module: bool
 
     # object_path is a .-delimited path to the object to execute, or a parent from which to infer the object
     # e.g.
     # function or local_entrypoint in module scope
     # app in module scope [+ method name]
     # app [+ function/entrypoint on that app]
-    object_path: str
+    object_path: str = dataclasses.field(default="")
 
 
-def parse_import_ref(object_ref: str) -> ImportRef:
+def parse_import_ref(object_ref: str, is_module: bool = False) -> ImportRef:
     if object_ref.find("::") > 1:
         file_or_module, object_path = object_ref.split("::", 1)
     elif object_ref.find(":") > 1:
         raise InvalidError(f"Invalid object reference: {object_ref}. Did you mean '::' instead of ':'?")
     else:
         file_or_module, object_path = object_ref, ""
 
-    return ImportRef(file_or_module, object_path)
+    return ImportRef(file_or_module, is_module, object_path)
 
 
 DEFAULT_APP_NAME = "app"
 
 
-def import_file_or_module(file_or_module: str):
+def import_file_or_module(import_ref: ImportRef, base_cmd: str = ""):
+    # TODO: Remove the base_cmd argument here once we have removed the deprecation
     if "" not in sys.path:
         # When running from a CLI like `modal run`
         # the current working directory isn't added to sys.path
         # so we add it in order to make module path specification possible
         sys.path.insert(0, "")  # "" means the current working directory
 
-    if file_or_module.endswith(".py"):
+    if not import_ref.file_or_module.endswith(".py") or import_ref.is_module:
```

Comment:
In what situations could the first condition be `True` and the second `False`? (i.e., a path that doesn't end in `.py` but is not a module?)

Reply:
That would be the case when users specify a module ref but don't send along the `-m` flag - i.e the case where we issue the warning below

Reply:
I see. It's maybe a little confusing when reading the code in this context that `import_ref.is_module` is data about how the user invoked Modal, not a statement about what sort of path we actually have.

Reply:
Yeah, maybe it should be called `load_as_module` or `use_module_mode`?